src/commands/hug.js

```diff
@@ -1,8 +1,66 @@
+import axios from 'axios';
+import Discord from 'discord.js';
 import DiscordCommand from './command.js';
 
 export default class HugCommand extends DiscordCommand {
-    async execute() {
-        const hug = this.getEmojiByName('HugSpin');
-        this.message.channel.send(`${hug}`);
+
+  async execute() {
+    if (!this.member) {
+      this.send("You must be in a Roo Bot enabled Discord server to use this command");
+      return;
+    }
+
+    const hug = this.getEmojiByName('HugSpin');
+    this.message.react(hug);
+
+    let receivers = "";
+    const mentionedUsernames = [];
+
+    if (this.hasRoleMentions || this.hasChannelMentions) {
+      this.send(`Please use \`\`\`${DiscordCommand.prefix}hug @user @user...\`\`\``);
+      return;
     }
+
+    if (this.hasUserMentions) {
```

Comment:
I would probably invert this conditional into a guard clause so we can unnest the conditionals

example code:

```js
const didHugSelf = !!(
  this.userMentions && 
  this.userMentions.size === 1 &&
  this.userMentions[0].id === this.message.author.id
);
if (!this.hasUserMentions || didHugSelf) {
  return await this.send(...) // themself
}

// now below you know `this.userMentions` has stuff more than themself
```